src/Interpreters/HashJoin.cpp

```diff
@@ -1158,21 +1159,150 @@ class AddedColumns
     std::vector<TypeAndName> type_name;
     MutableColumns columns;
     std::vector<size_t> right_indexes;
-    size_t lazy_defaults_count = 0;
     /// for ASOF
     const IColumn * left_asof_key = nullptr;
     Block sample_block;
+    std::vector<RowRef> lazy_right_columns;
```

Comment:
Could you please add comment why do we need vector of `lazy_right_columns` instead of defaults count as it used to be?